app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductActivity.java

```diff
@@ -275,8 +276,10 @@ private void setupViewPager( ViewPager viewPager )
 			adapterResult.addFragment( new NutritionProductFragment(), menuTitles[2] );
 			if( mState.getProduct().getNutriments() != null && mState.getProduct().getNutriments().contains(Nutriments.CARBON_FOOTPRINT) )
 			{
-				adapterResult.addFragment( new EnvironmentProductFragment(), menuTitles[4] );
-			}
+				adapterResult.addFragment( new EnvironmentProductFragment(), "Environment" );
+			} else if (mState.getProduct().getEnvironmentInfocard() != null && !mState.getProduct().getEnvironmentInfocard().isEmpty()) {
+			    adapterResult.addFragment(new EnvironmentProductFragment(), "Environment");
```

Comment:
Here we are using a `if-else if` statement , but it can be done in a single `if` statement like the other fragments. 
```Java
if( (mState.getProduct().getNutriments() != null && 
      mState.getProduct().getNutriments().contains(Nutriments.CARBON_FOOTPRINT) ||  
      (mState.getProduct().getEnvironmentInfocard() != null && !mState.getProduct().getEnvironmentInfocard().isEmpty()))
			{
                     adapterResult.addFragment(new EnvironmentProductFragment(), "Environment");
                         }
```

Reply:
Yes, I understand. I'll make the changes right away and push the commit. I only avoided a large conditional statement because I thought it would get confusing but splitting it over a couple of lines works out fine.